salt/modules/lxc.py

```diff
@@ -3191,24 +3191,21 @@ def bootstrap(name,
             if install:
                 rstr = __salt__['test.rand_str']()
                 configdir = '/tmp/.c_{0}'.format(rstr)
-                run(name,
-                    'install -m 0700 -d {0}'.format(configdir),
-                    python_shell=False)
+
+                cmd = 'install -m 0700 -d {0}'.format(configdir)
+                if run(name, cmd, python_shell=False):
+                    log.error('tmpdir {0} creation failed ({1}'
+                              .format(configdir, cmd))
+                    return False
+
                 bs_ = __salt__['config.gather_bootstrap_script'](
                     bootstrap=bootstrap_url)
-                dest_dir = os.path.join('/tmp', rstr)
-                for cmd in [
-                    'mkdir -p {0}'.format(dest_dir),
-                    'chmod 700 {0}'.format(dest_dir),
-                ]:
-                    if run_stdout(name, cmd):
-                        log.error(
-                            ('tmpdir {0} creation'
-                             ' failed ({1}').format(dest_dir, cmd))
-                        return False
-                cp(name,
-                   bs_,
-                   '{0}/bootstrap.sh'.format(dest_dir))
+                script = '/sbin/{0}_bootstrap.sh'.format(rstr)
```

Comment:
Yes, I think we may have made some mistakes here of course. But the cachedir is the best spot, go ahead and make a subdir in there if you need to, the value for the cachedir is ``__opts__['cachedir']``